src/Microsoft.SqlTools.ServiceLayer/AzureFunctions/GetAzureFunctionsOperation.cs

```diff
@@ -0,0 +1,71 @@
+﻿//
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+//
+
+using System;
+using System.IO;
+using System.Linq;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.SqlTools.ServiceLayer.AzureFunctions.Contracts;
+using Microsoft.SqlTools.Utility;
+using Microsoft.CodeAnalysis.CSharp;
+using System.Collections.Generic;
+
+namespace Microsoft.SqlTools.ServiceLayer.AzureFunctions
+{
+    /// <summary>
+    /// Class to represent getting the Azure Functions in a file
+    /// </summary>
+    class GetAzureFunctionsOperation
+    {
+        const string functionAttributeText = "FunctionName";
+
+        public GetAzureFunctionsParams Parameters { get; }
+
+        public GetAzureFunctionsOperation(GetAzureFunctionsParams parameters)
+        {
+            Validate.IsNotNull("parameters", parameters);
+            this.Parameters = parameters;
+        }
+
+        /// <summary>
+        /// Gets the names of all the azure functions in a file
+        /// </summary>
+        /// <returns>the result of trying to get the names of all the Azure functions in a file</returns>
+        public GetAzureFunctionsResult GetAzureFunctions()
+        {
+            try
+            {
+                string text = File.ReadAllText(Parameters.filePath);
+
+                SyntaxTree tree = CSharpSyntaxTree.ParseText(text);
+                CompilationUnitSyntax root = tree.GetCompilationUnitRoot();
+
+                // look for Azure Functions in the file
+                IEnumerable<AttributeArgumentSyntax> functionNameAttributes = from methodDeclaration in root.DescendantNodes().OfType<MethodDeclarationSyntax>()
+                                          where methodDeclaration.AttributeLists.Count > 0
+                                          where methodDeclaration.AttributeLists.Where(a => a.Attributes.Where(attr => attr.Name.ToString().Contains(functionAttributeText)).Count() == 1).Count() == 1
```

Comment:
Anything you can do to take this complex linq query and make it more readable?  Maybe break it up into a couple steps with comments or extract the `Where()` clauses into helper functions?

`First().ArgumentList.Arguments.First()).First()` is a particularly fun time, lol